src/core/reactor_backend.cc

```diff
@@ -1313,11 +1317,16 @@ class reactor_backend_uring final : public reactor_backend {
         }
         return sqe;
     }
-    future<> poll(pollable_fd_state& fd, int events) {
+
+    static bool is_speculated(pollable_fd_state& fd, int events) {
         if (events & fd.events_known) {
             fd.events_known &= ~events;
-            return make_ready_future<>();
+            return true;
         }
+        return false;
+    }
```

Comment:
Let's make this a member of pollable_fd_state for cleanliness.

The name should also indicate that it clears the speculation. Perhaps pollable_fd_state::get_and_clear_speculation(int events)?

Reply:
how about `take_speculation()`?

Reply:
ok, as long as it's explained.

Reply:
added comment to explain the API and the rationales.